src/Map.js

```diff
@@ -11,14 +11,6 @@ import _ from "lodash";
 import { getQueryStringValue, setQueryStringValue } from "./queryString";
 import Utils from "./Utils";
 
-/*
- * Expose a promise that resolves when the map is fully instantiated
- */
-let mapLoadedResolver;
```

Comment:
This was here to deal with the fact that loading the map js sdk is async (and thus the present of google.maps on the window object is async). 

Without this some of the other code that used google.map ... (especially the geometry functions) would fail.

It's possible your other changes make this not a problem though.

Reply:
Here's my understanding of the original behavior:
1. App mounts and some parts of the code waits for the Map promise to resolve.
2. Map mounts and initializes the Google Map and resolves the promise.
3. App calls "setTimeRange" and "updateToggleState" which calls Map functions that need the Map to be initialized first
4. Map does some update stuff

And the current behavior:
1. App mounts and sets state (including time range, toggle enabled/disabled) without calling any Map functions.
2. Map loads and initializes the Google Map.
3. Map detects time range, toggle props and does its stuff.

My own testing seems fine, but I wonder if there's a way to try to break this? Maybe add a delay in the map initialize?

Reply:
I think the original problem was with the dwell location visualization (where computing dwell locations tried to use window.google.maps before enabled).   With your changes (and explanation) I think that race is now fixed -- since the toggle isn't applied until after the map has been loaded.

Ship it!